### ISSUE TITLE
DialogResourceProviderFactoryImpl slows down bundle deployment

### ISSUE THREAD

**Issue by DavidAtWorking** [Adobe-Consulting-Services/acs-aem-commons#2211]
Same problem as closed issue [#1979](https://github.com/Adobe-Consulting-Services/acs-aem-commons/issues/1979)

It looks like since version 4.3, same issue comes back again. The code is in commit 0ffd40a class
DialogResourceProviderFactoryImpl.java put all the adapter factory register and unregister things back. Any particular reason to do this?

**Comment by joerghoh** [comment]
@justinedelson  as you have authored most of #2024, what do you think? For me it mostly looks like an merge error.

**Comment by henrykuijpers** [comment]
@justinedelson , it would be nice if this could be followed up - It is a very annoying issue. :(

**Comment by justinedelson** [comment]
Regarding https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/2024/files#diff-cb8fa962575c15cd6168356c5ba35191, this was done to avoid the single-argument use of `Class.forName` which is a known anti-pattern in OSGi (if not generally in Java). The standard pattern (at least that I'm aware of) to do this type of thing in AEM is via the `DynamicClassLoaderManager` which is available as an OSGi service. As a result, any `bind` invocations *prior* to the injection of this service need to be queued and then the actual class registration done after the component is activated (this could have also been done when the `DynamicClassLoaderManager` reference was injected, but it seems to me that doing it in the `activate` method is basically the same thing).

In other words, this change did not change *what* was done (`registerClass` is still invoked once per `AdapterFactory` before and after this change), but *when* it was done.

In the specific case described in #1979, it shouldn't really matter -- the step of "Redeploy bundle" seems to be described as happening *after* the `DialogResourceProviderFactoryImpl` component had been activated, at which point the code path (immediately calling `registerClass` upon binding of the `AdapterFactory`) would be the same before and after the change in https://github.com/Adobe-Consulting-Services/acs-aem-commons/commit/0ffd40a15460f8be8d3591e17d5648184f63e42f

**Comment by henrykuijpers** [comment]
To me, this does not look good:
```
10.03.2020 20:46:05.076 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234619, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:05.079 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234618, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:05.080 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236013, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:05.087 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236014, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:05.089 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:09.562 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234621, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:09.564 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234620, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:09.566 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236014, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:09.573 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236015, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:09.576 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:14.249 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234623, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:14.251 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234622, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:14.253 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236015, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:14.260 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236016, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:14.263 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:18.950 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234625, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:18.953 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234624, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:18.954 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236016, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:18.960 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236017, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:18.963 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:23.581 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234627, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:23.583 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234626, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:23.585 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236017, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:23.591 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236018, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:23.593 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:28.268 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234629, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:28.270 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234628, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:28.272 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236018, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:28.279 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236019, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:28.281 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:32.963 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234631, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:32.965 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234630, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:32.967 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236019, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:32.973 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236020, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:32.976 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:37.673 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234633, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:37.675 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234632, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:37.677 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236020, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:37.683 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236021, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:37.686 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:42.548 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234635, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:42.551 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234634, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:42.552 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236021, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:42.559 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236022, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:42.561 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:47.261 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234637, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:47.263 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234636, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:47.265 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236022, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:47.272 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236023, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:47.275 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:52.361 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234639, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:52.365 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234638, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:52.368 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236023, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:52.378 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236024, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:52.381 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
10.03.2020 20:46:58.675 *INFO* [OsgiInstallerImpl] org.apache.sling.adapter Service [234641, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
10.03.2020 20:46:58.679 *INFO* [OsgiInstallerImpl] org.apache.sling.models.impl Service [234640, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:58.681 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236024, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent UNREGISTERING
10.03.2020 20:46:58.689 *INFO* [OsgiInstallerImpl] com.adobe.acs.acs-aem-commons-bundle Service [com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl,236025, [com.adobe.acs.commons.mcp.DialogResourceProviderFactory]] ServiceEvent REGISTERED
10.03.2020 20:46:58.692 *INFO* [OsgiInstallerImpl] com.day.cq.preferences.PreferencesServiceFactory This service is no longer functional (PreferencesServiceFactory).
```

This is even with 4.2.2.

**Comment by davidjgonzalez** [comment]
@henrykuijpers any chance you could try it out w/ the attached (I'm running on a Mid 2011 MacBook air today as my other laptop stopped powering on, so I don't have much on this computer to test with)

[acs-aem-commons-ui.apps-4.4.3-SNAPSHOT-min.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/4318959/acs-aem-commons-ui.apps-4.4.3-SNAPSHOT-min.zip)

**Comment by davidjgonzalez** [comment]
@henrykuijpers  / @DavidAtWorking  any chance to test the codebase w/ the possible fix?

**Comment by davidjgonzalez** [comment]
@henrykuijpers / @DavidAtWorking any chance you were able to try this SNAPSHOT out?

**Comment by DavidAtWorking** [comment]
> @henrykuijpers / @DavidAtWorking any chance you were able to try this SNAPSHOT out?

I have check your package.
It does not work for me.

**Comment by DavidAtWorking** [comment]
@davidjgonzalez

**Comment by joerghoh** [comment]
I cannot reproduce this here on AEM 6.5.2 with the current master build (I have added some INFO messages to the bind/unbind methods to indicate that these are called instead of the activate/deactivate of the service).

```
11.04.2020 13:13:37.568 *INFO* [qtp717971973-5333] com.adobe.cq.sample.we.retail.core BundleEvent STOPPING
...
11.04.2020 13:13:37.660 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22593, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.664 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22594, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.671 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.678 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22596, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.678 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22598, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.681 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.682 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22601, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.682 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22602, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.685 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.686 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22606, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.686 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22607, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.688 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.689 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22611, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.690 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22612, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
...
11.04.2020 13:13:37.737 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22646, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.739 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.740 *INFO* [qtp717971973-5333] org.apache.sling.models.impl Service [22650, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.740 *INFO* [qtp717971973-5333] org.apache.sling.adapter Service [22652, [org.apache.sling.adapter.Adaption]] ServiceEvent UNREGISTERING
11.04.2020 13:13:37.745 *INFO* [qtp717971973-5333] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl unbind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:37.747 *INFO* [qtp717971973-5333] com.adobe.cq.sample.we.retail.core BundleEvent STOPPED
11.04.2020 13:13:38.814 *INFO* [qtp717971973-5520] com.adobe.cq.sample.we.retail.core BundleEvent STARTING
11.04.2020 13:13:38.814 *INFO* [qtp717971973-5520] com.adobe.cq.sample.we.retail.core BundleEvent STARTED
...
11.04.2020 13:13:38.859 *INFO* [qtp717971973-5520] org.apache.sling.models.impl Service [24885, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent REGISTERED
11.04.2020 13:13:38.871 *INFO* [qtp717971973-5520] org.apache.sling.adapter Service [24886, [org.apache.sling.adapter.Adaption]] ServiceEvent REGISTERED
11.04.2020 13:13:38.874 *INFO* [qtp717971973-5520] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl bind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:38.875 *INFO* [qtp717971973-5520] org.apache.sling.models.impl Service [24887, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent REGISTERED
11.04.2020 13:13:38.876 *INFO* [qtp717971973-5520] org.apache.sling.adapter Service [24888, [org.apache.sling.adapter.Adaption]] ServiceEvent REGISTERED
11.04.2020 13:13:38.878 *INFO* [qtp717971973-5520] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl bind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:38.879 *INFO* [qtp717971973-5520] org.apache.sling.models.impl Service [24889, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent REGISTERED
11.04.2020 13:13:38.880 *INFO* [qtp717971973-5520] org.apache.sling.adapter Service [24890, [org.apache.sling.adapter.Adaption]] ServiceEvent REGISTERED
11.04.2020 13:13:38.882 *INFO* [qtp717971973-5520] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl bind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:38.883 *INFO* [qtp717971973-5520] org.apache.sling.models.impl Service [24891, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent REGISTERED
11.04.2020 13:13:38.883 *INFO* [qtp717971973-5520] org.apache.sling.adapter Service [24892, [org.apache.sling.adapter.Adaption]] ServiceEvent REGISTERED
...
11.04.2020 13:13:38.955 *INFO* [qtp717971973-5520] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl bind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:38.956 *INFO* [qtp717971973-5520] org.apache.sling.models.impl Service [24915, [org.apache.sling.api.adapter.AdapterFactory]] ServiceEvent REGISTERED
11.04.2020 13:13:38.957 *INFO* [qtp717971973-5520] org.apache.sling.adapter Service [24916, [org.apache.sling.adapter.Adaption]] ServiceEvent REGISTERED
11.04.2020 13:13:38.961 *INFO* [qtp717971973-5520] com.adobe.acs.commons.mcp.impl.DialogResourceProviderFactoryImpl bind called with org.apache.sling.models.impl.ModelAdapterFactory@5cb454f
11.04.2020 13:13:40.938 *INFO* [sling-default-5] com.day.cq.wcm.mobile.core.impl.devicespecs.WurflXmlLoader Checking if wurfl.xml needs parsing...
```

**Comment by davidjgonzalez** [comment]
Closing as we cannot reproduce and have no received any further feedback on this issue from the reporting users. Please re-open w/ more details on how to re-produce if needed.

**Comment by joerghoh** [comment]
I think that the "slowness" of any bundle deployment might also be caused by other services, which use the ResourceChangeListener and are not restricted to Resources only, but also react to registering and unregistering of ResourceProviders. Because the DialogResourceProviderFactoryImpl also registers a new ResourceProvider for each SlingModel, this can significant overhead. 
I fixed some of these occurrences in #2265, but also AEM is far from perfect (https://experienceleaguecommunities.adobe.com/t5/adobe-experience-manager/disabling-adobe-granite-maintenance-scheduler-update-when/m-p/358491/highlight/true#M25610).

So I encourage everyone to apply the change I mentioned in the link and try again the latest version of ACS AEM Commons.
